modules/data.atmosphere/R/download.ICOS.R

```diff
@@ -0,0 +1,233 @@
+#' Download ICOS Ecosystem data products
+#'
+#' Currently available products: 
+#' Drought-2018 ecosystem eddy covariance flux product https://www.icos-cp.eu/data-products/YVR0-4898
+#' ICOS Final Fully Quality Controlled Observational Data (Level 2) https://www.icos-cp.eu/data-products/ecosystem-release
+#'
+#'
+#' @param sitename ICOS id of the site. Example - "BE-Bra"
+#' @param outfolder path to the directory where the output file is stored. If specified directory does not exists, it is created.
+#' @param start_date start date of the data request in the form YYYY-MM-DD
+#' @param end_date end date area of the data request in the form YYYY-MM-DD
+#' @param product ICOS product to be downloaded. Currently supported options: "Drought2018", "ETC"
+#' @param overwrite should existing files be overwritten. Default False.
+#' @return information about the output file
+#' @export
+#' @examples
+#' \dontrun{
+#' download.ICOS("FI-Sii", "/home/carya/pecan", "2016-01-01", "2018-01-01", product="Drought2018")
+#' }
+#' @author Ayush Prasad
+#'
+download.ICOS <-
+  function(sitename,
+           outfolder,
+           start_date,
+           end_date,
+           product,
+           overwrite = FALSE) {
+    download_file_flag <- TRUE
+    extract_file_flag <- TRUE
+    
+    
+    if (tolower(product) == "drought2018") {
+      # construct output CSV file name
+      output_file_name <-
+        paste0(
+          "FLX_",
+          sitename,
+          "_FLUXNET2015_FULLSET_HH_",
+          as.character(format(as.Date(start_date), '%Y')),
+          "-2018_beta-3.csv"
+        )
+      
+      # construct zip file name
+      zip_file_name <-
+        paste0(outfolder, "/Drought", sitename, ".zip")
+      
+      # data type, can be found from the machine readable page of the product
+      data_type <-
+        "http://meta.icos-cp.eu/resources/cpmeta/dought2018ArchiveProduct"
+      
+      file_name <-
+        paste0('FLX_', sitename, '_FLUXNET2015_FULLSET_HH')
+      
+      format_name <- "ICOS_Drought2018_HH"
+      
+    } else if (tolower(product) == "etc") {
+      output_file_name <-
+        paste0(sitename, "/ICOSETC_", sitename, "_FLUXNET_HH_01.csv")
+      
+      # construct zip file name
+      zip_file_name <-
+        paste0(outfolder, "/ICOSETC_Archive_", sitename, ".zip")
+      
+      # data type, can be found from the machine readable page of the product
+      data_type <-
+        "http://meta.icos-cp.eu/resources/cpmeta/etcArchiveProduct"
+      
+      file_name <-
+        paste0(sitename, "/ICOSETC_", sitename, "_FLUXNET_HH")
+      
+      format_name <- "ICOS_ETC_HH"
```

Comment:
Is this fine? The ecosystem archive product is similar to drought except for 2-3 variables, if we were subsetting the variables, I think we could've used the same format. The other option I can think of is to have the same format but to subset the variables in `met2CF.ICOS` so that the column numbers remain same,

Reply:
I think this is fine. Not-subsetting variables increases your function's usability outside of PEcAn, so I'm liking it and defining two formats is not a big deal. Do the variables of our interest happen to have the same column numbers by any chance (in the raw files I mean)? :)

Like you said, using a single format and handling the mapping in the `met2CF.ICOS` can also be an option. Actually, [met2CF.AmerifluxLBL](https://github.com/PecanProject/pecan/blob/develop/modules/data.atmosphere/R/met2CF.AmerifluxLBL.R#L44) does something similar. Normally I wouldn't like this (as it sort of obscures what is on the DB record versus what is being processed) but I suspect that the variable names are so similar that we may get away with one format for both ICOS products. Shall we give this a try?

Reply:
> Do the variables of our interest happen to have the same column numbers by any chance (in the raw files I mean)? 

Oh yes, I just saw that the variables required for met2cf have the same column numbers. I think only the constraint data variables differ like NEE_VUT_REF is 98 in the ecosystem archive but 87 in drought 2018. I guess we can have the same format now without making any changes.


![Screenshot from 2021-07-05 22-51-58](https://user-images.githubusercontent.com/11568631/124503726-c6597c00-dde3-11eb-9198-14ea4c6cfd01.png)